src/FileIndexer.ts

```diff
@@ -25,22 +28,52 @@ export class FileIndexer {
   private localCounter = new Counter()
   private propertyCounters: Map<string, Counter> = new Map()
   private localSymbolTable: Map<ts.Node, LsifSymbol> = new Map()
+  private workingDirectoryRegExp: RegExp
   constructor(
     public readonly checker: ts.TypeChecker,
+    public readonly options: ProjectOptions,
     public readonly input: Input,
     public readonly document: lsif.lib.codeintel.lsiftyped.Document,
     public readonly globalSymbolTable: Map<ts.Node, LsifSymbol>,
     public readonly packages: Packages,
     public readonly sourceFile: ts.SourceFile
-  ) {}
+  ) {
+    this.workingDirectoryRegExp = new RegExp(options.cwd, 'g')
+  }
   public index(): void {
+    this.emitSourceFileOccurrence()
     this.visit(this.sourceFile)
   }
+  private emitSourceFileOccurrence(): void {
+    const symbol = this.lsifSymbol(this.sourceFile)
+    if (symbol.isEmpty()) {
+      return
+    }
+    this.document.occurrences.push(
+      new lsif.lib.codeintel.lsiftyped.Occurrence({
+        range: [0, 0, 0],
+        symbol: symbol.value,
+        symbol_roles: lsiftyped.SymbolRole.Definition,
+      })
+    )
+    const moduleName =
+      this.sourceFile.moduleName ||
+      path
+        .basename(this.sourceFile.fileName)
+        .replace(/.tsx?$/, '')
+        .replace(/.jsx?/, '')
```

Comment:
1. Is it ever possible that we have files with the same base name but with different extensions? IIUC this happens with translation/conversion, but we will not index the generated files.
2. I think you need to add `.d.ts`, `.cjs` and `.mjs` extensions too.
3. Since the `.` is inside the regex, it needs escaping, right? Or is this some reason why it doesn't need to be escaped here?

I'm wondering, instead of dropping the extension here, can we get the filename and include the extension when emitting the occurrence for the import instead? Any reason to prefer dropping the extension instead of doing that?

Reply:
This case should not happen since it's a fallback when `moduleName` is undefined, and I'm guessing it's always defined for `ts.SourceFile`

Reply:
My preference for dropping the extension was to best-effort mirror the output of `moduleName`, even if it doesn't handle all cases.

Reply:
Hmm, is the replacement even working? The occurrences seem to have the file extensions...

Reply:
Turns out `moduleName` was in fact undefined almost everywhere! I updated the logic to keep the file extension so that we don't treat d.ts/mjs/cjs extensions differently